api/api.go

```diff
@@ -66,43 +117,44 @@ func (srv *Server) initAPI() {
 	// Gateway API Calls
 	if srv.gateway != nil {
 		router.GET("/gateway", srv.gatewayHandler)
-		router.POST("/gateway/add/:netaddress", srv.gatewayAddHandler)
-		router.POST("/gateway/remove/:netaddress", srv.gatewayRemoveHandler)
+		router.POST("/gateway/add/:netaddress", requirePassword(srv.gatewayAddHandler, password))
+		router.POST("/gateway/remove/:netaddress", requirePassword(srv.gatewayRemoveHandler, password))
 	}
 
 	// Host API Calls
 	if srv.host != nil {
 		router.GET("/host", srv.hostHandlerGET)
-		router.POST("/host", srv.hostHandlerPOST)
-		router.POST("/host/announce", srv.hostAnnounceHandler)
-		router.POST("/host/delete/:filecontractid", srv.hostDeleteHandler)
+		router.POST("/host", requirePassword(srv.hostHandlerPOST, password))
+		router.POST("/host/announce", requirePassword(srv.hostAnnounceHandler, password))
+		router.POST("/host/delete/:filecontractid", requirePassword(srv.hostDeleteHandler, password))
 	}
 
 	// Miner API Calls
 	if srv.miner != nil {
 		router.GET("/miner", srv.minerHandler)
-		router.GET("/miner/header", srv.minerHeaderHandlerGET)
-		router.POST("/miner/header", srv.minerHeaderHandlerPOST)
-		router.GET("/miner/start", srv.minerStartHandler)
-		router.GET("/miner/stop", srv.minerStopHandler)
-		router.GET("/miner/headerforwork", srv.minerHeaderHandlerGET)  // COMPATv0.4.8
-		router.POST("/miner/submitheader", srv.minerHeaderHandlerPOST) // COMPATv0.4.8
+		router.GET("/miner/header", requirePassword(srv.minerHeaderHandlerGET, password))
+		router.POST("/miner/header", requirePassword(srv.minerHeaderHandlerPOST, password))
+		router.GET("/miner/start", requirePassword(srv.minerStartHandler, password))
+		router.GET("/miner/stop", requirePassword(srv.minerStopHandler, password))
+		// TODO: doesn't adding authentication break compatibility?
+		router.GET("/miner/headerforwork", requirePassword(srv.minerHeaderHandlerGET, password))  // COMPATv0.4.8
+		router.POST("/miner/submitheader", requirePassword(srv.minerHeaderHandlerPOST, password)) // COMPATv0.4.8
```

Comment:
makes sense to me. Especially considering how badly we broke compatibility with 060, I don't see much of a reason to keep these endpoints around.